Zeroconf/NetworkInterface.cs

```diff
@@ -225,18 +233,18 @@ Task ListenForAnnouncementsAsync(System.Net.NetworkInformation.NetworkInterface
                     socket.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.AddMembership, multOpt);
 
 
-                    cancellationToken.Register((() =>
-                                                {
-                                                    ((IDisposable)client).Dispose();
```

Comment:
If we remove this, what happens if cancellation is called when we're awaiting in client.ReceiveAsync(). Because that receive call doesn't take the token, it seems like we'd never break out of that loop without hard-canceling things via the Dispose on the client tied to the token?

Reply:
cause client is surround by using statement. so it should be released after leaving the scope

Reply:
The issue is that it'd not leaving scope because the RecieveAsync call never returns.

To repro, start ZeroconfTest.Netcore and click "Browse". It never returns the data to the UX.